### PR TITLE
20464 Cleanup SystemVersion and SystemVersionTest (but stay compatible)

### PR DESCRIPTION
**Pull request by astares**
- remove unnecessary instance variable "version"
- introduce explicit instance variable "build"
- add parsing via Regex (in #fromVersionString:)
- deprecate old, unused and wrong parsing methods
- calculate the highestUpdate when not set from the outside
- proper categorization
- fix class comments
- add many new tests, also tests that should ensure compatibility

### CODE REVIEW

**Comment by astares** [comment]
https://pharo.fogbugz.com/f/cases/20464/Cleanup-SystemVersion-and-SystemVersionTest-but-stay-compatible

**Comment by astares** [comment]
Before this can go into the image the bootstrap/src/Pharo30Bootstrap.package/PBImageBuilder50.class/instance/initializeImage.st needs to be changed to only give "alpha" as a suffix.

And after this PR is merged into the development image we need to set the build number explicitly into the new ivar "build".

**Comment by astares** [comment]
At least one of the checks will fail (by intention) - we first need to change the initializeImage.st 
as written above to only give the  "alpha" as a suffix.

**Comment by astares** [comment]
OK - so SmalltalkImage>>#lastUpdateString was now changed. AboutBox is fixed now too.
I guess/hope there is nothing left so far - time to merge

**Comment by tesonep** [comment]
The execution in the CI is in a endless loop when trying to load RPackage. Please check the problem because it affects the infrastructure.

**Comment by astares** [comment]
I can not check it - at least I dont know how.

**Comment by astares** [comment]
OK - anyone able to review the files?

File: src/System-Support.package/SystemVersion.class/instance/initialize.st [hunk 1]
```diff
@@ -1,4 +1,10 @@
 initialization
 initialize
 	super initialize.
+	type := 'Pharo'.
+	major := 0.
+	minor := 0.	 	
+	suffix := ''.
+	build := -1. "means no build number given, as we also accept a build number 0"
```
Comment (pavel-krivanek) [inline src/System-Support.package/SystemVersion.class/instance/initialize.st]: invalidBuildNumber should be used here. Then the comment is not required

Comment (astares) [reply src/System-Support.package/SystemVersion.class/instance/initialize.st]: Fixed it

**Pull request merged**